### PR TITLE
Feature/88 point cloud triangulation

### PR DESCRIPTION
**Pull request by JulianHammerl**
* **Tickets addressed:** Confluence 88
* **Review:** By commit  
* **Merge strategy:** Merge (no squash)  

## Description

A module was created that estimates the location of features (creates a point cloud) by triangulation, given a two camera locations and corresponding images. The commits are organized as follows:

- Commit 1: Create necessary message definitions
- Commit 2: Create module
- Commit 3: UnitTest
- Commit 4: Documentation

## Verification

A UnitTest was created that compares the generated point cloud with the true point cloud.

## Documentation

This is a new module. An RST documentation was added.

## Future work

The triangulation() method of the module will eventually be moved from the module to an external utilities library. Thus, it is kept general here: Even though this module assumes that all images come from the same camera, the triangulation() method is set up such that the camera calibration matrix and camera direction cosine matrix (DCM) could be different for each image.

### CODE REVIEW

File: src/architecture/msgPayloadDefC/DirectionOfMotionMsgPayload.h [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+//@endcond
+{
+    bool valid; //!< --  Quality of measurement
+    uint64_t cameraID; //!< --  Camera who's motion is in question
+    uint64_t timeOfDirectionEstimate; //!< --  time of the direction of motion computation
+    double v_C_hat[3]; //!< -- [-]   Camera direction of motion
+    double covar_C[3*3]; //!< -- [-]   Unvertainty in direction of motion
+}DirectionOfMotionMsgPayload;
+
+#endif /* DIRECTION_MOTION_MSG_H */
```
Comment (thibaudteil) [inline src/architecture/msgPayloadDefC/DirectionOfMotionMsgPayload.h]: This message will be on develop (very) shortly, I think you'll want to remove this commit before rebasing and pushing

File: src/fswAlgorithms/pointCloudProcessing/pointCloudTriangulation/pointCloudTriangulation.cpp [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+    double pX = 2.*tan(fieldOfView*resolutionX/resolutionY/2.0);
+    double pY = 2.*tan(fieldOfView/2.0);
+    double dX = resolutionX/pX;
+    double dY = resolutionY/pY;
+    double up = resolutionX/2;
+    double vp = resolutionY/2;
+    // build inverse K^-1 of camera calibration matrix K
+    this->cameraCalibrationMatrixInverse << 1./dX, -alpha/(dX*dY), (alpha*vp - dY*up)/(dX*dY),
+                                            0., 1./dY, -vp/dY,
+                                            0., 0., 1.;
```
Comment (thibaudteil) [inline src/fswAlgorithms/pointCloudProcessing/pointCloudTriangulation/pointCloudTriangulation.cpp:177]: We should consider refactoring some of these commonly used methods alongside some of the dynamics/integration refactor. For another time

**Review by thibaudteil — COMMENTED** [review]
Good module and test

File: src/architecture/msgPayloadDefC/DirectionOfMotionMsgPayload.h (hunk 1 above)
Comment (JulianHammerl) [reply src/architecture/msgPayloadDefC/DirectionOfMotionMsgPayload.h]: Done.

File: src/fswAlgorithms/pointCloudProcessing/pointCloudTriangulation/pointCloudTriangulation.cpp (hunk 2 above)
Comment (JulianHammerl) [reply src/fswAlgorithms/pointCloudProcessing/pointCloudTriangulation/pointCloudTriangulation.cpp:177]: Agreed!

**Review by thibaudteil — APPROVED** [review]

**Pull request merged**